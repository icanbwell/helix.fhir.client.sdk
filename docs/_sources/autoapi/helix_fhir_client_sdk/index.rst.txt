:py:mod:`helix_fhir_client_sdk`
===============================

.. py:module:: helix_fhir_client_sdk


Subpackages
-----------
.. toctree::
   :titlesonly:
   :maxdepth: 3

   exceptions/index.rst
   filters/index.rst
   graph/index.rst
   loggers/index.rst
   responses/index.rst
   validators/index.rst


Submodules
----------
.. toctree::
   :titlesonly:
   :maxdepth: 1

   fhir_client/index.rst
   well_known_configuration/index.rst


