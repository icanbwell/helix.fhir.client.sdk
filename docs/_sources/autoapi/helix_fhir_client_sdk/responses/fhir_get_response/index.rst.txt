:py:mod:`helix_fhir_client_sdk.responses.fhir_get_response`
===========================================================

.. py:module:: helix_fhir_client_sdk.responses.fhir_get_response


Module Contents
---------------

Classes
~~~~~~~

.. autoapisummary::

   helix_fhir_client_sdk.responses.fhir_get_response.FhirGetResponse




.. py:class:: FhirGetResponse(url, responses, error, access_token)

   
   Class that encapsulates the response from FHIR server

   :param url: url that was being accessed
   :param responses: response text
   :param error: Any error returned by FHIR server
   :param access_token: access token that was used


