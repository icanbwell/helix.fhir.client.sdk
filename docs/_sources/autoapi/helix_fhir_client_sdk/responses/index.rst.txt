:py:mod:`helix_fhir_client_sdk.responses`
=========================================

.. py:module:: helix_fhir_client_sdk.responses


Submodules
----------
.. toctree::
   :titlesonly:
   :maxdepth: 1

   fhir_get_response/index.rst
   fhir_merge_response/index.rst


