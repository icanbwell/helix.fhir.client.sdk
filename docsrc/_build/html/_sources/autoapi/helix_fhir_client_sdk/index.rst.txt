:py:mod:`helix_fhir_client_sdk`
===============================

.. py:module:: helix_fhir_client_sdk


Subpackages
-----------
.. toctree::
   :titlesonly:
   :maxdepth: 3

   exceptions/index.rst
   loggers/index.rst
   responses/index.rst
   validators/index.rst


Submodules
----------
.. toctree::
   :titlesonly:
   :maxdepth: 1

   fhir_client/index.rst


