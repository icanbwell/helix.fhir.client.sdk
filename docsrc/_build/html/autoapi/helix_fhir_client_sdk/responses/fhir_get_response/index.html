

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>helix_fhir_client_sdk.responses.fhir_get_response &mdash; helix.fhir.client.sdk  documentation</title>
  

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/graphviz.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js"></script>
        <script src="../../../../_static/jquery.js"></script>
        <script src="../../../../_static/underscore.js"></script>
        <script src="../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> helix.fhir.client.sdk
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../../index.html">API Reference</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">helix.fhir.client.sdk</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li><code class="xref py py-mod docutils literal notranslate"><span class="pre">helix_fhir_client_sdk.responses.fhir_get_response</span></code></li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../../../_sources/autoapi/helix_fhir_client_sdk/responses/fhir_get_response/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-helix_fhir_client_sdk.responses.fhir_get_response">
<span id="helix-fhir-client-sdk-responses-fhir-get-response"></span><h1><a class="reference internal" href="#module-helix_fhir_client_sdk.responses.fhir_get_response" title="helix_fhir_client_sdk.responses.fhir_get_response"><code class="xref py py-mod docutils literal notranslate"><span class="pre">helix_fhir_client_sdk.responses.fhir_get_response</span></code></a><a class="headerlink" href="#module-helix_fhir_client_sdk.responses.fhir_get_response" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this headline">¶</a></h2>
<div class="section" id="classes">
<h3>Classes<a class="headerlink" href="#classes" title="Permalink to this headline">¶</a></h3>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#helix_fhir_client_sdk.responses.fhir_get_response.FhirGetResponse" title="helix_fhir_client_sdk.responses.fhir_get_response.FhirGetResponse"><code class="xref py py-obj docutils literal notranslate"><span class="pre">FhirGetResponse</span></code></a></p></td>
<td><p>Class that encapsulates the response from FHIR server</p></td>
</tr>
</tbody>
</table>
<dl class="py class">
<dt class="sig sig-object py" id="helix_fhir_client_sdk.responses.fhir_get_response.FhirGetResponse">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">helix_fhir_client_sdk.responses.fhir_get_response.</span></span><span class="sig-name descname"><span class="pre">FhirGetResponse</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">responses</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">error</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">access_token</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#helix_fhir_client_sdk.responses.fhir_get_response.FhirGetResponse" title="Permalink to this definition">¶</a></dt>
<dd><p>Class that encapsulates the response from FHIR server</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>url</strong> (<em>str</em>) – url that was being accessed</p></li>
<li><p><strong>responses</strong> (<em>str</em>) – response text</p></li>
<li><p><strong>error</strong> (<em>Optional</em><em>[</em><em>str</em><em>]</em>) – Any error returned by FHIR server</p></li>
<li><p><strong>access_token</strong> (<em>Optional</em><em>[</em><em>str</em><em>]</em>) – access token that was used</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</div>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Helix Team.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>