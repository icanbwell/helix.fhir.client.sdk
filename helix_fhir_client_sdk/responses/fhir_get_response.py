from typing import Optional


class FhirGetResponse:
    def __init__(
        self,
        url: str,
        responses: str,
        error: Optional[str],
        access_token: Optional[str],
        total_count: Optional[int],
        status: int,
    ) -> None:
        """
        Class that encapsulates the response from FHIR server

        :param url: url that was being accessed
        :param responses: response text
        :param error: Any error returned by FHIR server
        :param access_token: access token that was used
        :param total_count: count of total records that match the provided query.
                            Only set if include_total_count was set to avoid expensive operation by server.
        """
        self.url: str = url
        self.responses: str = responses
        self.error: Optional[str] = error
        self.access_token: Optional[str] = access_token
        self.total_count: Optional[int] = total_count
        self.status: int = status
