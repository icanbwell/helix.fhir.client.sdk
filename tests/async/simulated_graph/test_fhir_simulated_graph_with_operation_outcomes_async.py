import json
from pathlib import Path
from typing import Dict, Any

from mockserver_client.mockserver_client import (
    MockServerFriendlyClient,
    mock_request,
    mock_response,
    times,
)

from helix_fhir_client_sdk.fhir_client import FhirClient

from helix_fhir_client_sdk.responses.fhir_get_response import FhirGetResponse


async def test_fhir_simulated_graph_with_operation_outcomes_async() -> None:
    print("")
    data_dir: Path = Path(__file__).parent.joinpath("./")
    graph_json: Dict[str, Any]
    with open(data_dir.joinpath("graphs").joinpath("aetna.json"), "r") as file:
        contents = file.read()
        graph_json = json.loads(contents)

    test_name = "test_fhir_simulated_graph_with_operation_outcomes_async"

    mock_server_url = "http://mock-server:1080"
    mock_client: MockServerFriendlyClient = MockServerFriendlyClient(
        base_url=mock_server_url
    )

    relative_url: str = test_name
    absolute_url: str = mock_server_url + "/" + test_name

    mock_client.clear(f"/{test_name}/*.*")
    mock_client.reset()

    response_text = {
        "resourceType": "Patient",
        "id": "1",
        "generalPractitioner": [{"reference": "Practitioner/5"}],
        "managingOrganization": {"reference": "Organization/6"},
    }

    mock_client.expect(
        mock_request(path=f"/{relative_url}/Patient/1", method="GET"),
        mock_response(body=response_text),
        timing=times(1),
    )

    response_text = {"resourceType": "Practitioner", "id": "5"}
    mock_client.expect(
        mock_request(path=f"/{relative_url}/Practitioner/5", method="GET"),
        mock_response(body=response_text),
        timing=times(1),
    )

    response_text = {"resourceType": "Organization", "id": "6"}
    mock_client.expect(
        mock_request(path=f"/{relative_url}/Organization/6", method="GET"),
        mock_response(body=response_text),
        timing=times(1),
    )

    response_text = {"entry": [{"resource": {"resourceType": "Coverage", "id": "7"}}]}
    mock_client.expect(
        mock_request(
            path=f"/{relative_url}/Coverage",
            querystring={"patient": "1"},
            method="GET",
        ),
        mock_response(body=response_text),
        timing=times(1),
    )

    response_text = {
        "entry": [
            {"resource": {"resourceType": "Observation", "id": "8"}},
            {
                "resource": {
                    "resourceType": "OperationOutcome",
                    "issue": [
                        {
                            "severity": "warning",
                            "code": "suppressed",
                            "details": {
                                "coding": [
                                    {
                                        "system": "urn:oid:1.2.840.114350.1.13.0.1.7.2.657369",
                                        "code": "59204",
                                        "display": "The authenticated client's search request applies to a sub-resource that the client is not authorized for. Results of this sub-type will not be returned.",
                                    }
                                ],
                                "text": "The authenticated client's search request applies to a sub-resource that the client is not authorized for. Results of this sub-type will not be returned.",
                            },
                            "diagnostics": "Client not authorized for DocumentReference - Document Information. Search results of this type have not been included.",
                        }
                    ],
                }
            },
        ]
    }
    mock_client.expect(
        mock_request(
            path=f"/{relative_url}/Observation",
            querystring={
                "patient": "1",
                "category": "vital-signs,social-history,laboratory",
            },
            method="GET",
        ),
        mock_response(body=response_text),
        timing=times(1),
    )

    fhir_client = FhirClient()
    fhir_client = fhir_client.expand_fhir_bundle(False)

    fhir_client.extra_context_to_return({"service_slug": "medstar"})

    auth_access_token = "my_access_token"
    if auth_access_token:
        fhir_client = fhir_client.set_access_token(auth_access_token)

    fhir_client = fhir_client.url(absolute_url).resource("Patient")
    response: FhirGetResponse = await fhir_client.simulate_graph_async(
        id_="1",
        graph_json=graph_json,
        contained=False,
        separate_bundle_resources=False,
    )
    print(response.responses)

    expected_json = {
        "entry": [
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Patient/1",
                },
                "resource": {
                    "generalPractitioner": [{"reference": "Practitioner/5"}],
                    "id": "1",
                    "managingOrganization": {"reference": "Organization/6"},
                    "resourceType": "Patient",
                },
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Practitioner/5",
                },
                "resource": {"id": "5", "resourceType": "Practitioner"},
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Organization/6",
                },
                "resource": {"id": "6", "resourceType": "Organization"},
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Coverage?patient=1",
                },
                "resource": {"id": "7", "resourceType": "Coverage"},
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory",
                },
                "resource": {"id": "8", "resourceType": "Observation"},
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/ExplanationOfBenefit?patient=1",
                },
                "resource": {
                    "issue": [
                        {
                            "code": "not-found",
                            "details": {
                                "coding": [
                                    {
                                        "code": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/ExplanationOfBenefit?patient=1",
                                        "system": "https://www.icanbwell.com/url",
                                    },
                                    {
                                        "code": "ExplanationOfBenefit",
                                        "system": "https://www.icanbwell.com/resourceType",
                                    },
                                    {
                                        "code": 404,
                                        "system": "https://www.icanbwell.com/statuscode",
                                    },
                                    {
                                        "code": "my_access_token",
                                        "system": "https://www.icanbwell.com/accessToken",
                                    },
                                ]
                            },
                            "diagnostics": '{"url": '
                            '"http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/ExplanationOfBenefit?patient=1", '
                            '"error": "NotFound", '
                            '"status": 404, '
                            '"extra_context_to_return": '
                            '{"service_slug": '
                            '"medstar"}, "accessToken": '
                            '"my_access_token", '
                            '"requestId": null, '
                            '"resourceType": '
                            '"ExplanationOfBenefit", '
                            '"id": null}',
                            "severity": "error",
                        }
                    ],
                    "resourceType": "OperationOutcome",
                },
                "response": {"status": "404"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory",
                },
                "resource": {
                    "issue": [
                        {
                            "code": "suppressed",
                            "details": {
                                "coding": [
                                    {
                                        "code": "59204",
                                        "display": "The "
                                        "authenticated "
                                        "client's "
                                        "search "
                                        "request "
                                        "applies "
                                        "to a "
                                        "sub-resource "
                                        "that "
                                        "the "
                                        "client "
                                        "is not "
                                        "authorized "
                                        "for. "
                                        "Results "
                                        "of "
                                        "this "
                                        "sub-type "
                                        "will "
                                        "not be "
                                        "returned.",
                                        "system": "urn:oid:1.2.840.114350.1.13.0.1.7.2.657369",
                                    },
                                    {
                                        "code": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory",
                                        "system": "https://www.icanbwell.com/url",
                                    },
                                    {
                                        "code": "Observation",
                                        "system": "https://www.icanbwell.com/resourceType",
                                    },
                                    {
                                        "code": 200,
                                        "system": "https://www.icanbwell.com/statuscode",
                                    },
                                    {
                                        "code": "my_access_token",
                                        "system": "https://www.icanbwell.com/accessToken",
                                    },
                                ],
                                "text": "The authenticated "
                                "client's search "
                                "request applies to a "
                                "sub-resource that the "
                                "client is not "
                                "authorized for. "
                                "Results of this "
                                "sub-type will not be "
                                "returned.",
                            },
                            "diagnostics": "Client not authorized for "
                            "DocumentReference - "
                            "Document Information. "
                            "Search results of this "
                            "type have not been "
                            "included.",
                            "severity": "warning",
                        }
                    ],
                    "resourceType": "OperationOutcome",
                },
                "response": {"status": "200"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationRequest?patient=1",
                },
                "resource": {
                    "issue": [
                        {
                            "code": "not-found",
                            "details": {
                                "coding": [
                                    {
                                        "code": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationRequest?patient=1",
                                        "system": "https://www.icanbwell.com/url",
                                    },
                                    {
                                        "code": "MedicationRequest",
                                        "system": "https://www.icanbwell.com/resourceType",
                                    },
                                    {
                                        "code": 404,
                                        "system": "https://www.icanbwell.com/statuscode",
                                    },
                                    {
                                        "code": "my_access_token",
                                        "system": "https://www.icanbwell.com/accessToken",
                                    },
                                ]
                            },
                            "diagnostics": '{"url": '
                            '"http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationRequest?patient=1", '
                            '"error": "NotFound", '
                            '"status": 404, '
                            '"extra_context_to_return": '
                            '{"service_slug": '
                            '"medstar"}, "accessToken": '
                            '"my_access_token", '
                            '"requestId": null, '
                            '"resourceType": '
                            '"MedicationRequest", "id": '
                            "null}",
                            "severity": "error",
                        }
                    ],
                    "resourceType": "OperationOutcome",
                },
                "response": {"status": "404"},
            },
            {
                "request": {
                    "method": "GET",
                    "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationDispense?patient=1",
                },
                "resource": {
                    "issue": [
                        {
                            "code": "not-found",
                            "details": {
                                "coding": [
                                    {
                                        "code": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationDispense?patient=1",
                                        "system": "https://www.icanbwell.com/url",
                                    },
                                    {
                                        "code": "MedicationDispense",
                                        "system": "https://www.icanbwell.com/resourceType",
                                    },
                                    {
                                        "code": 404,
                                        "system": "https://www.icanbwell.com/statuscode",
                                    },
                                    {
                                        "code": "my_access_token",
                                        "system": "https://www.icanbwell.com/accessToken",
                                    },
                                ]
                            },
                            "diagnostics": '{"url": '
                            '"http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/MedicationDispense?patient=1", '
                            '"error": "NotFound", '
                            '"status": 404, '
                            '"extra_context_to_return": '
                            '{"service_slug": '
                            '"medstar"}, "accessToken": '
                            '"my_access_token", '
                            '"requestId": null, '
                            '"resourceType": '
                            '"MedicationDispense", '
                            '"id": null}',
                            "severity": "error",
                        }
                    ],
                    "resourceType": "OperationOutcome",
                },
                "response": {"status": "404"},
            },
        ]
    }

    bundle = json.loads(response.responses)
    assert bundle == expected_json
