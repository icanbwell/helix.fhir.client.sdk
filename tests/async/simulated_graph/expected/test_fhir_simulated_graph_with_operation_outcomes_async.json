{
  "type": "collection",
  "resourceType": "Bundle",
  "total": 6,
  "entry": [
    {
      "resource": {
        "resourceType": "Patient",
        "id": "1",
        "generalPractitioner": [
          {
            "reference": "Practitioner/5"
          }
        ],
        "managingOrganization": {
          "reference": "Organization/6"
        }
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Patient/1",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    },
    {
      "resource": {
        "resourceType": "Practitioner",
        "id": "5"
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Practitioner/5",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    },
    {
      "resource": {
        "resourceType": "Organization",
        "id": "6"
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Organization/6",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    },
    {
      "resource": {
        "resourceType": "Coverage",
        "id": "7"
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Coverage?patient=1",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    },
    {
      "resource": {
        "resourceType": "Observation",
        "id": "8"
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    },
    {
      "resource": {
        "resourceType": "OperationOutcome",
        "issue": [
          {
            "severity": "warning",
            "code": "suppressed",
            "details": {
              "coding": [
                {
                  "system": "urn:oid:1.2.840.114350.1.13.0.1.7.2.657369",
                  "code": "59204",
                  "display": "The authenticated client's search request applies to a sub-resource that the client is not authorized for. Results of this sub-type will not be returned."
                },
                {
                  "system": "https://www.icanbwell.com/url",
                  "code": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory"
                },
                {
                  "system": "https://www.icanbwell.com/resourceType",
                  "code": "Observation"
                },
                {
                  "system": "https://www.icanbwell.com/statuscode",
                  "code": 200
                },
                {
                  "system": "https://www.icanbwell.com/accessToken",
                  "code": "my_access_token"
                }
              ],
              "text": "The authenticated client's search request applies to a sub-resource that the client is not authorized for. Results of this sub-type will not be returned."
            },
            "diagnostics": "Client not authorized for DocumentReference - Document Information. Search results of this type have not been included."
          }
        ]
      },
      "request": {
        "url": "http://mock-server:1080/test_fhir_simulated_graph_with_operation_outcomes_async/Observation?patient=1&category=vital-signs,social-history,laboratory",
        "method": "GET"
      },
      "response": {
        "status": "200"
      }
    }
  ]
}
